src/bandersnatch/tests/test_package.py

```diff
@@ -43,3 +45,18 @@ async def test_package_not_found(caplog: CaptureFixture, master: Master) -> None
     with pytest.raises(PackageNotFound):
         await package.update_metadata(master)
     assert "foo no longer exists on PyPI" in caplog.text
+
+
+@pytest.mark.asyncio
+async def test_package_update_metadata_gives_up_after_3_timeouts(
+    caplog: CaptureFixture, master: Master
+) -> None:
+    master.get_package_metadata = asynctest.CoroutineMock(  # type: ignore
+        side_effect=TimeoutError
+    )
+    package = Package("foo", serial=11)
+
+    with pytest.raises(ConnectionTimeout):
+        await package.update_metadata(master, attempts=3)
+    assert master.get_package_metadata.await_count == 3  # type: ignore
+    assert "not updating. Giving up" in caplog.text
```

Comment:
assert “Connection timeout for” in str(ConnectionTimeout())

This could help you hit your __str__() method